libs/article/api/src/lib/article-revision-dataloader.service.ts

```diff
@@ -19,45 +19,20 @@ export class ArticleRevisionDataloaderService implements Primeable<RevisionMap>
 
       for (const articleId of articleIds) {
         revisionPromises.push(
-          this.prisma.articleRevision.findFirst({
-            where: {
-              articleId
-              // on purpose not checking for publishedAt
-              // so we only get the latest revision, we check for it later on
-            },
-            orderBy: {
-              createdAt: 'desc'
-            }
-          }),
-          this.prisma.articleRevision.findFirst({
-            where: {
-              articleId,
-              publishedAt: {
-                gt: new Date()
-              }
-            },
-            orderBy: {
-              createdAt: 'desc'
-            }
-          }),
-          this.prisma.articleRevision.findFirst({
+          this.prisma.articleRevision.findMany({
```

Comment:
@elias-summermatter could be further optimized by using raw queries. Do you think it's worth doing it in this step or should we make a ticket as an improvement?

Going from 1+n to 1+1 in terms of queries would be nearly always a win I am guessing.